code/modules/mob/living/carbon/xenomorph/castes/ravager/abilities_ravager.dm

```diff
@@ -233,3 +233,156 @@
+	ability_name = "Rage"
+	plasma_cost = 0 //We're limited by cooldowns, not plasma
+	cooldown_timer = 60 SECONDS
+	keybind_flags = XACT_KEYBIND_USE_ABILITY | XACT_IGNORE_SELECTED_ABILITY
+	keybind_signal = COMSIG_XENOABILITY_RAGE
+	///Determines the power of Rage's many effects. Power scales inversely with the Ravager's HP; min 0.25 at 50% of Max HP, max 1 while in negative HP. 0.5 and above triggers especial effects.
+	var/rage_power
+	///Determines the Sunder to impose when Rage ends
+	var/rage_sunder
+	///Determines the Plasma to remove when Rage ends
+	var/rage_plasma
+
+/datum/action/xeno_action/rage/on_cooldown_finish()
+	to_chat(owner, "<span class='xenodanger'>We are able to enter our rage once again.</span>")
+	owner.playsound_local(owner, 'sound/effects/xeno_newlarva.ogg', 25, 0, 1)
+	return ..()
+
+/datum/action/xeno_action/rage/can_use_action(atom/A, silent = FALSE, override_flags)
+	. = ..()
+	if(!.)
+		return FALSE
+
+	var/mob/living/carbon/xenomorph/ravager/rager = owner
+
+	var/rage_health_threshold = rager.maxHealth * 0.5 //Need to be at 50% of max hp or lower to rage
+	if(rager.health > rage_health_threshold)
+		if(!silent)
+			to_chat(rager, "<span class='xenodanger'>Our health isn't low enough to rage! We must take [rager.health - rage_health_threshold] more damage!</span>")
+		return FALSE
+
+
+/datum/action/xeno_action/rage/action_activate()
+	var/mob/living/carbon/xenomorph/ravager/X = owner
+
+	rage_power = (1-(X.health/X.maxHealth)) * 0.5 //Calculate the power of our rage; scales with difference between current and max HP
+
+	if(X.health < 0) //Gain additional rage with negative HP; gain + 0.0075 rage power per point of negative HP
+		rage_power += X.health * -0.0075
+
+	rage_power = min(1, rage_power) //Cap rage power so that we don't get way too insane.
+
+	var/rage_power_radius = CEILING(rage_power * 7, 1) //Define radius of the SFX
+
+	X.visible_message("<span class='danger'>\The [X] becomes frenzied, bellowing with a shuddering roar!</span>", \
+	"<span class='highdanger'>We bellow as our fury overtakes us! RIP AND TEAR!</span>")
+	X.do_jitter_animation(1000)
+
+
+	//Roar SFX; volume scales with rage
+	playsound(X.loc, 'sound/voice/alien_roar2.ogg', clamp(100 * rage_power, 25, 80), 0)
+
+	var/bonus_duration
+	if(rage_power > 0.5) //If we're super pissed it's time to get crazy
+		var/datum/action/xeno_action/charge = X.actions_by_path[/datum/action/xeno_action/activable/charge]
+		var/datum/action/xeno_action/ravage = X.actions_by_path[/datum/action/xeno_action/activable/ravage]
+		if(charge)
+			charge.clear_cooldown() //Reset charge cooldown
+		if(ravage)
+			ravage.clear_cooldown() //Reset ravage cooldown
+
+		for(var/mob/living/witness in hearers(rage_power_radius, X)) //Roar that applies cool SFX and some soft CC
```

Comment:
ew a bit.

First, you are doing twice hearers, this proc is bad and should be minimized (especially since you have to istype every item)

Second, consider using cheap_get_human_near and cheap_get_xeno_near rather than hearers, AFAIK it's more optimized